frontend/src/app/application-forms/temporary-outfitters/temp-outfitter-left-nav.component.spec.ts

```diff
@@ -0,0 +1,170 @@
+    spyOn(document, 'getElementById').and.callFake(function() {
+      return {
+        value: 'test',
+        getBoundingClientRect() {
+          return { top: 19 };
+        }
+      };
+    });
+    component.track(new Event('scroll'));
+    expect(component.top).toEqual('40px');
+    expect(component.position).toEqual('fixed');
+  });
+
+  it('should set absolute position if top of container is greator than 20px', () => {
+    spyOn(document, 'getElementById').and.callFake(function() {
+      return {
+        value: 'test',
+        getBoundingClientRect() {
+          return { top: 20 };
+        }
+      };
+    });
+    spyOn(window, 'innerHeight').and.callFake(function() {
+      return 50;
+    });
+    component.track(new Event('scroll'));
+    expect(component.top).toEqual('250px');
+    expect(component.position).toEqual('absolute');
+  });
+
+  it(
+    'should set current section',
+    inject([UtilService], util => {
+      component.gotoHashtag('test', new Event('click'));
+      expect(util.currentSection).toEqual('test-section');
+    })
+  );
+
+  it('should update current section on changes', () => {
+    component.currentSection = 'section-guide-identification';
+    component.ngOnChanges();
+    expect(component.applicationForm.controls.guideIdentification.touched).toBeTruthy();
+    component.currentSection = 'section-operating-plan';
+    component.ngOnChanges();
+    expect(component.applicationForm.controls.operatingPlan.touched).toBeTruthy();
+    component.currentSection = 'section-acknowledgement-of-risk';
+    component.ngOnChanges();
+    expect(component.applicationForm.controls.acknowledgementOfRisk.touched).toBeTruthy();
+    component.currentSection = 'section-experience';
+    component.ngOnChanges();
+    expect(
+      component.applicationForm.controls.tempOutfitterFields.controls.experienceFields.controls
+        .listAllNationalForestPermits.touched
+    ).toBeTruthy();
+  });
+
+  it('should get control status', () => {
+    const control = component.applicationForm.get('guideIdentification');
+    expect(component.getControlStatus(control)).toBe('ng-untouched');
+    control.markAsTouched();
+    expect(component.getControlStatus(control)).toBe('ng-valid');
+    const control2 = component.applicationForm.get('liabilityInsurance');
+    control2.markAsTouched();
+    expect(component.getControlStatus(control2)).toBe('ng-invalid');
+  });
+
+  it('should get group status', () => {
+    const group = component.applicationForm.get('tempOutfitterFields');
+    expect(component.getGroupStatus(group, false)).toBe('ng-untouched');
+    group.markAsTouched();
+    expect(component.getGroupStatus(group, false)).toBe('ng-valid');
+    const group2 = component.applicationForm.get('applicantInfo');
+    group.markAsTouched();
+    expect(component.getGroupStatus(group2, true)).toBe('ng-invalid');
+  });
+});
+
+class MockUtilService {
```

Comment:
Does this need to be mocked or can you pull the actual `util` service?

Reply:
The test I was using it for wasn't very helpful, so i removed it and the mock